torch/fx/proxy.py

```diff
@@ -529,6 +529,24 @@ def find_tracer(a):
                                        name=tracer.graph._target_to_str(orig_method.__name__))
 
 
+@compatibility(is_backward_compatible=False)
+class MetaProxy(Proxy):
+    """
+    A Proxy subclass that propagates metadata (meta['val']) during graph tracing.
+    """
+
+    def __repr__(self) -> str:
+        return f'MetaProxy({self.node.name})'
+
+    @classmethod
+    def __torch_function__(cls, orig_method, types, args=None, kwargs=None):
+        args = args if args else ()
+        kwargs = kwargs if kwargs else {}
+
+        proxy = super().__torch_function__(orig_method, types, args, kwargs)
+        proxy.node.meta["val"] = orig_method(*[a.node.meta["val"] if isinstance(a, Proxy) else a for a in args], **kwargs)
```

Comment:
It might be safer to explicitly enable the fake tensor mode here before doing the prop. This saves you in a situation where MetaProxy is representing a non-Tensor argument (which can happen with plain Proxy usage, although it doesn't happen for your code, I think), and consequently you end up doing a function call without any fake tensor inputs, and then you end up allocating a real tensor. This is probably most conveniently done by explicitly associating a fake mode with the proxy.